src/index.js

```diff
@@ -159,9 +159,15 @@ class UglifyJsPlugin {
                 task.cacheKey = serialize({
                   'uglify-es': versions.uglify,
                   'uglifyjs-webpack-plugin': versions.plugin,
+                  'source-map': versions.sourceMap,
                   'uglifyjs-webpack-plugin-options': this.options,
                   path: compiler.outputPath ? `${compiler.outputPath}/${file}` : file,
-                  hash: crypto.createHash('md5').update(input).digest('hex'),
+                  inputHash: crypto.createHash('md5').update(
+                    input ? input.toString() : '',
+                  ).digest('hex'),
+                  inputSourceMapHash: crypto.createHash('md5').update(
```

Comment:
@michael-ciniawsky for right invalidate cache - yes, we have source map(s) (which can changed in difference versions) from webpack, after we use `source-map` for generate map(s) and `source-map` also can change source map(s)

Reply:
Can you give an example please ? I don't get it :)

Reply:
@michael-ciniawsky 
1. We have `webpack` and build our apps, uglify plugin cached all.
2. We upgrade `webpack` (which change source map algorithm, example update `source-map`), now we have other `map` value in `const { source, map } = asset.sourceAndMap();`, but we don't use `inputSourceMap` as part of cache key => we have invalid source map => we should invalidate cache
3. We upgrade `source-map`, but we don't use `source-map` as part of cache key => we will get invalid source map => we should invalidate cache when `source-map` version changed

Reply:
I still don't think it's worth the overhead compared to how often this might occur, at some point someone simply needs to clean the cache if there is a conflict for time to time. The `x` package versions is already debatable imho. Maybe better to get other opinions on this :)

Reply:
@michael-ciniawsky In recent times, I personally encountered this problem, I think vrjatli scarcely except for us will look at it :smile: Also right invalidate cache is very important

Reply:
> In recent times, I personally encountered this problem

Yep because `source-map` got an update for the first time in months, it definitely is bug when this happens I'm not disagreeing with that fact nor that i t is inconvenient, but the solution is not ideal atm imho

> I think vrjatli scarcely except for us will look at it 😄

I can't summon them aswell :) and it's definitely a problem in general e.g if others think the perf hit is acceptable here, I'm not objection against it at all

Reply:
@michael-ciniawsky The invalidation will occur only after updating the source maps, this is extremely rare, but it is worthwhile to avoid this situation, so I propose to accept this PR :smile: